### PR TITLE
typst: add `updateScript`, run tests

### PR DESCRIPTION
**Pull request by Misterio77**
## Description of changes

Add `updateScript` to typst, and also run the package tests [as instructed by upstream](https://github.com/typst/typst/tree/main/tests#running-the-tests).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
That's a duplicate of #312469, it would be nice to keep the tests and update parts though.

**Comment by Misterio77** [comment]
Upstream integration tests capture the CLI args, and defines [`--num-threads`](pkgs/build-support/rust/hooks/cargo-check-hook.sh). So I added a `postPatch` step to replace it with `--test-threads`, as passed by [`cargo-check-hook.sh`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/hooks/cargo-check-hook.sh#L39)

**Comment by Misterio77** [comment]
> That's a duplicate of #312469, it would be nice to keep the tests and update parts though.

Ah my bad! Didn't see it. Will drop the bump commit and rebase, then.

File: pkgs/by-name/ty/typst/package.nix [hunk 1]
```diff
@@ -46,13 +47,23 @@ rustPlatform.buildRustPackage rec {
     OPENSSL_NO_VENDOR = true;
   };
 
+  # Fix for "Found argument '--test-threads' which wasn't expected, or isn't valid in this context"
+  postPatch = ''
```
Comment (drupol) [inline pkgs/by-name/ty/typst/package.nix:51]: Is this something we can submit upstream ?

Comment (Misterio77) [reply pkgs/by-name/ty/typst/package.nix:51]: Hmmm maybe. But I wonder if it'd be better if we fixed this in nixpkgs instead? It seems to be a quite frequent issue:

```
$ grep -r "Found argument '--test-threads'"
pkgs/applications/misc/navi/default.nix:    # error: Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/applications/version-management/glitter/default.nix:  # error: Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/servers/search/sonic-server/default.nix:  # Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/tools/misc/lsd/default.nix:  # Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/tools/misc/sfz/default.nix:  # error: Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/tools/text/tidy-viewer/default.nix:  # error: Found argument '--test-threads' which wasn't expected, or isn't valid in this context
pkgs/by-name/ty/typst/package.nix:  # Fix for "Found argument '--test-threads' which wasn't expected, or isn't valid in this context"
```

There's apparently a [`RUST_TEST_THREADS` environment variable](https://manpages.debian.org/bullseye/rustc/rustc.1.en.html#RUST_TEST_THREADS), maybe we should use it instead?

Comment (drupol) [reply pkgs/by-name/ty/typst/package.nix:51]: yeah let's try

Comment (Misterio77) [reply pkgs/by-name/ty/typst/package.nix:51]: Here's a draft PR: https://github.com/NixOS/nixpkgs/pull/312505

This should target staging, right? As probably causes rebuild for every single rust package.

Comment (drupol) [reply pkgs/by-name/ty/typst/package.nix:51]: Yes indeed.

Comment (drupol) [reply pkgs/by-name/ty/typst/package.nix:51]: Let's merge this PR and refactor it when #312505 has landed.

**Pull request merged**